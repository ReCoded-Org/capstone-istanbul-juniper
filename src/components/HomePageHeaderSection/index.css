.homePageHeaderContainer__title {
  text-align: center;
  display: inline-block;
  font-family: "salsa";
  font-size: clamp(0.5rem, 20px, 4vw);
  display: none;
}

.homePageHeaderContainer__img {
  display: none;
}

.homePageHeaderContainer__paragraph {
  font-size: 15px;
  font-family: "Raleway";
  display: none;
}

.homePageHeaderContainer {
  padding-top: 150px;
  font-size: 2vw;
  background: #b4d0c4;
}

.homePageHeaderContainer__row {
  border: none;
  background: #f8ac4c;
  margin: auto;
  width: 100%;
  font-family: "Salsa", cursive;
  font-weight: 550;
  transition: all 0.5s;
  padding: 5px;
  font-size: clamp(0.5rem, 15px, 4vw);
  cursor: pointer;
  min-height: 130%;
  outline: none;
  color: black;
}

.homePageHeaderContainer__button {
  border-radius: 20px;
  border: none;
  background: #f8ac4c;
  margin: auto;
  width: 100%;
  font-family: "Salsa", cursive;
  font-weight: 550;
  transition: all 0.5s;
  padding: 5px;
  font-size: clamp(0.5rem, 15px, 4vw);
  cursor: pointer;
  min-height: 130%;
  display: none;
}

.homePageHeaderContainer__button:hover {
  transform: scale(1.2);
  transition: all 0.5s;
}

/* Media query rules to override previous rules, as necessary */
/* when the breakpoint happend we will see the one image only without the title and paragraph */
@media screen and (min-width: 1000px) {
  .homePageHeaderContainer__img {
    display: block;
  }
  .homePageHeaderContainer__title {
    display: block;
  }
  .homePageHeaderContainer__paragraph {
    display: block;
  }
  .homePageHeaderContainer__button {
    display: block;
  }
}

@media screen and (max-width: 1000px) {
  .homePageHeaderContainer__rightImg {
    width: 350px;
  }
}

@media screen and (max-width: 800px) {
  .homePageHeaderContainer__rightImg {
    width: 550px;
  }
}

@media screen and (max-width: 400px) {
  .homePageHeaderContainer__rightImg {
    width: 330px;
  }
}
