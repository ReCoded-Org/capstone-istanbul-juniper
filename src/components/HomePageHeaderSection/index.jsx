import React from "react";
import "antd/dist/antd.css";
import { Row, Col } from "antd";
import { useTranslation } from "react-i18next";
import i18next from "i18next";
import { Link } from "react-router-dom";
import homePageHeaderImgLeft from "../../images/child.png";
import homePageHeaderImgRight from "../../images/family.png";
import "./index.css";

export default function HomePageHeaderSection() {
  const { t } = useTranslation();
  const currentLanguage = i18next.language;
  return (
    <>
      <div className="homePageHeaderContainer">
        <Row justify="space-around" align="middle">
          <Col span={4}>
            <img
              alt="children playing in the garden"
              src={`${
                currentLanguage === "ar"
                  ? homePageHeaderImgLeft
                  : homePageHeaderImgRight
              }`}
              className="homePageHeaderContainer__rightImg"
            />
          </Col>

          <Col span={4} offset={6}>
            <Row>
              <h2
                className={`${
                  currentLanguage === "ar"
                    ? "homePageHeaderContainer__arTitle"
                    : "homePageHeaderContainer__title"
                }`}
              >
                {t("home.headerSection.mainText")}
              </h2>
              <br />
              <p className="homePageHeaderContainer__paragraph">
                {t("home.headerSection.secondaryText")}
              </p>
            </Row>
            <Row justify="center" className="homePageHeaderContainer__button">
              <Col>
                <Link to="/games">
                  <button className="homePageHeaderContainer__row">
                    {t("home.headerSection.headerButton")}
                  </button>
                </Link>
              </Col>
            </Row>
          </Col>
          <Col offset={6} span={4}>
            <img
              className="homePageHeaderContainer__img"
              alt="children playing in the garden"
              src={`${
                currentLanguage === "ar"
                  ? homePageHeaderImgRight
                  : homePageHeaderImgLeft
              }`}
            />
          </Col>
        </Row>
      </div>
    </>
  );
}
