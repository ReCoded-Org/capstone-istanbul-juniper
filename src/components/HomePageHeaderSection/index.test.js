import React from "react";
import HomePageHeaderSection from "./index.jsx";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

it("HomePageHeaderSection snapshot test", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <HomePageHeaderSection />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
