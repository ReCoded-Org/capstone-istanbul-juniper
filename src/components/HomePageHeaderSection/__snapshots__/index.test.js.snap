// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageHeaderSection snapshot test 1`] = `
<div
  className="homePageHeaderContainer"
>
  <div
    className="ant-row ant-row-space-around ant-row-middle"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-4"
      style={Object {}}
    >
      <img
        alt="children playing in the garden"
        className="homePageHeaderContainer__rightImg"
        src="family.png"
      />
    </div>
    <div
      className="ant-col ant-col-4 ant-col-offset-6"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={Object {}}
      >
        <h2
          className="homePageHeaderContainer__title"
        >
          home.headerSection.mainText
        </h2>
        <br />
        <p
          className="homePageHeaderContainer__paragraph"
        >
          home.headerSection.secondaryText
        </p>
      </div>
      <div
        className="ant-row ant-row-center homePageHeaderContainer__button"
        style={Object {}}
      >
        <div
          className="ant-col"
          style={Object {}}
        >
          <a
            href="/games"
            onClick={[Function]}
          >
            <button
              className="homePageHeaderContainer__row"
            >
              home.headerSection.headerButton
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-4 ant-col-offset-6"
      style={Object {}}
    >
      <img
        alt="children playing in the garden"
        className="homePageHeaderContainer__img"
        src="child.png"
      />
    </div>
  </div>
</div>
`;
